syntax = "proto3";
package example;

import "google/api/annotations.proto";
import "google/rpc/code.proto";

option go_package = "example/grpc/proto";
option java_package = "org.example.grpc.proto";
option java_outer_classname = "HelloProto";

service HelloService {
    rpc SayHello (HelloRequest) returns (HelloResponse) {
        option (google.api.http) = {
            post: "/hello"
            body: "*"
        };
    }
}

message HelloRequest {
    string your_name = 1;
}

message HelloResponse {
    google.rpc.Code code = 1;
    string message = 2;
}
